### PR TITLE
Add DIPK to the suite

### PR DESCRIPTION
**Pull request by JudithBernett**
Todo: 
- flake8
- remove optuna
- write tests

### CODE REVIEW

**Comment by JudithBernett** [comment]
I have corrected flake8 and mypy errors for DIPK. After going through the code, I have some questions/remarks: 
1. The number of heads was never used; it couldn't be set via the hyperparameters file, and it just defaulted to 1. I have changed that such that it can be set over the hyperparamters.yaml. Was there a particular reason for this?
2. The forward layer of the predictor and the forward layer of the AttentionLayer use "g", which seems to be the whole torch_geometric.data.data.Data object containing x, GEF, and BNF, which are also passed separately as arguments. This seems unnecessary.
3. What happened to the GraphEncoder? It was never used but is still commented out in the code and also the parameter "embedding_dim" that it needs as input is still part of the Predictor init
4. There was a wrong indentation of "epoch_loss /= it + 1" in the training loop in dipk.py such that it was never correctly calculated, I think. Is this correct? I have put it inside the loop now.

**Comment by PascalIversen** [comment]
1. In the paper they described using multi-head attention, but in the code the number of heads was set to one. I think it is a good idea to try higher numbers and make it a hpam! They did not report the number in the paper
2. good point, agreed!
3. The graph autoencoder is used to encode the drugs. Currently, we provide the encoded drug representations directly for the model. (Still need to upload it to Zenodo) However, now that I think about it... It would be cleaner to call it in the train/predict functions, so that we can apply it to new data. (Similar issue for the gene expression autoencoder, which is pretrained on the gene expression data and then subsequently applied to encode the gene expression, currently we just directly load those encoded gene expression vectors, also makes the model transductive.....) 
4. no i think this is supposed to be after the loop. It normalizes the loss. But it is confusing that they use the iterator for that, should be len(train_loader) or something. But it is not printed anyway? I think we can get rid of it
5. / 6. will do asap :D

**Review by JudithBernett — COMMENTED** [review]
looks great, approved!
